include/alpaka/math/cbrt/CbrtUniformCudaHipBuiltIn.hpp

```diff
@@ -30,22 +31,18 @@ namespace alpaka
         {
             //! The CUDA cbrt trait specialization.
             template<typename TArg>
-            struct Cbrt<CbrtUniformCudaHipBuiltIn, TArg, std::enable_if_t<std::is_arithmetic<TArg>::value>>
+            struct Cbrt<CbrtUniformCudaHipBuiltIn, TArg, std::enable_if_t<std::is_arithmetic_v<TArg>>>
             {
                 __device__ auto operator()(CbrtUniformCudaHipBuiltIn const& cbrt_ctx, TArg const& arg)
                 {
                     alpaka::ignore_unused(cbrt_ctx);
-                    return ::cbrt(arg);
-                }
-            };
 
-            template<>
-            struct Cbrt<CbrtUniformCudaHipBuiltIn, float>
-            {
-                __device__ auto operator()(CbrtUniformCudaHipBuiltIn const& cbrt_ctx, float const& arg) -> float
-                {
-                    alpaka::ignore_unused(cbrt_ctx);
-                    return ::cbrtf(arg);
+                    if constexpr(is_decayed_v<TArg, float>)
+                        return ::cbrtf(arg);
+                    else if constexpr(is_decayed_v<TArg, double> || std::is_integral_v<TArg>)
+                        return ::cbrt(arg);
```

Comment:
I am not sure, doesn't integer part cause a conversion warning? If so, should be separated with explicit conversion there

Reply:
So far the CI hasn't complained about this part, so I guess the conversion is fine.

Reply:
I don't think we check it in unit tests. We should at least try it manually and see it compiles

Reply:
Wow, you're right. We only check `float` and `double`. That sounds dangerous...

Reply:
I added the following check to one of our test cases:

```c++
auto c = alpaka::math::cbrt(acc, 49);
ALPAKA_CHECK(*success, c >= 7.0);
```

Ignore the check, I just wanted to make sure the compiler doesn't optimize away the unused `c`. It doesn't complain about a conversion, though.

Reply:
I think the previous CUDA/HIP specializations did not support float and integral as a mix, since they were partially specializd either specifically for these two cases or floating-point types.

Thanks for checking